### ISSUE TITLE
Automatically, annoyingly removes spaces as you type with codi.vim

### ISSUE THREAD

**Issue by dylan-chong** [thirtythreeforty/lessspace.vim#12]
As you type into the buffer when Codi is started, Codi evaluates the code. Unfortunately, a side effect of this behaviour is that this plug-in decides to strip the white space as you type - as you are still in insert mode. This is rather annoying, especially when you are trying to type code that is indented.

To illustrate this behaviour, see this:

![ezgif-1-d3a9dd2f53](https://user-images.githubusercontent.com/17482349/42132285-03889fea-7d69-11e8-85e0-3cc6ff38e014.gif)

I have mentioned this in this issue here https://github.com/metakirby5/codi.vim/issues/90

**Comment by thirtythreeforty** [comment]
Hey thanks for opening an issue.  Sorry you're having trouble.  Unfortunately it looks like both pieces of software are working as intended...

I'm having trouble completely understanding the Codi docs, but it looks like your best option is to change the [`opt_autocmd` option](https://github.com/metakirby5/codi.vim/blob/26fdafe24d24c6c1ff45eab7d8d14e1a9072a26c/doc/codi.txt#L235) to `InsertLeave`.  I _think_ this will trigger Codi when you leave Insert mode, rather than on every text change.  LessSpace only runs when you leave Insert mode or when you make a change in normal mode, so that might fix the symptom you're hitting.

LessSpace has a mechanism to work around conflicting plugins—if you're curious, search the source code for "temporary," although it isn't a documented API right now.  But I don't think it would be useful in this case, because Codi [sometimes doesn't call the `CodiUpdatePost`](https://github.com/metakirby5/codi.vim/blob/26fdafe24d24c6c1ff45eab7d8d14e1a9072a26c/autoload/codi.vim#L352-L355) autocommand that I would otherwise use to stop LessSpace from updating.  I'd be happy to hear suggestions about how to make this work.

Another option may just be adding an autocommand or blacklist to disable LessSpace in Codi buffers.

Sorry I don't have an immediate band-aid.  What would work best for you?

**Comment by thirtythreeforty** [comment]
Actually, upon further reading, it looks like the `CodiUpdatePre` and `CodiUpdatePost` commands may work.  Can you try running the following either in a vimrc or in the Vim command line?

```
augroup lessspace_codi
  autocmd!
  autocmd User CodiUpdatePre call lessspace#TemporaryDisableBegin()
  autocmd User CodiUpdatePost call lessspace#TemporaryDisableEnd()
augroup END
```

**Comment by dylan-chong** [comment]
> it looks like your best option is to change the opt_autocmd option to InsertLeave.

Yup this works for me, only problem is that when you make modifications in normal mode (for example x, or pasting) Codi does not get updated


> Actually, upon further reading, it looks like the CodiUpdatePre and CodiUpdatePost commands may work. Can you try running the following either in a vimrc or in the Vim command line?

I have just tried this, but less space does not seem to strip the white space at all

**Comment by thirtythreeforty** [comment]
> does not seem to strip the white space at all

Yeah I was afraid of that.  I think Codi never calls the Post hook, because it's async.

That's an interesting normal mode problem. I would think that would Just Work.  I will download Codi in a day or two when I have some time and see what I can come up with.

**Comment by thirtythreeforty** [comment]
Oh, duh... my bad: Try setting the `opt_autocmd` option to `CursorHold`. Since it only triggers when you leave insert mode, obviously it will never trigger when you do normal mode changes.

**Comment by dylan-chong** [comment]
I tested it without setting the autocmd (so using default setting)
